src/support/scale.rs

```diff
@@ -75,6 +75,19 @@ impl<const BITS: usize, const LIMBS: usize> CompactAs for CompactUint<BITS, LIMB
     }
 }
 
+impl<const BITS: usize, const LIMBS: usize> MaxEncodedLen for CompactUint<BITS, LIMBS> {
+    fn max_encoded_len() -> usize {
+        Uint::<BITS, LIMBS>::max_encoded_len()
+    }
+}
```

Comment:
🤔 these max_encoded_len values seems to be off,
I would have expected this test to pass
```rust
    #[test]
    fn test_scale_compact_max_len() {
        const_for!(BITS in [1, 2, 3, 7, 8, 9, 15, 16, 17, 29, 30, 31, 32, 33, 63, 64, 65, 127, 128, 129, 256, 384, 512, 535] {
            const LIMBS: usize = nlimbs(BITS);
            let value: CompactUint<BITS, LIMBS> = Uint::<BITS, LIMBS>::MAX.into();
            assert!(CompactUint::<BITS, LIMBS>::max_encoded_len(), value.encode().len());
        });
    }
```